src/integrations/GoogleCloudPubSubIntegration.ts

```diff
@@ -0,0 +1,336 @@
+                    const parentSpan = tracer.getActiveSpan();
+
+                    const subscription = request.subscription;
+                    const operationName = `PULL-${subscription}`;
+
+                    ThundraLogger.debug(`<GoogleCloudPubSubIntegration> Starting pull span with name ${operationName}`);
+
+                    const getMessageBody = (receivedMessages: any []) => {
+                        if (!receivedMessages) {
+                            return;
+                        }
+
+                        const messages = receivedMessages.map((receivedMessage) => receivedMessage.message);
+                        return GooglePubSubUtils.parseMessages(messages);
+                    };
+
+                    const getProjectId = () => {
+                        if (!subscription) {
+                            return;
+                        }
+
+                        const topicInfo = subscription.split('/');
+                        if (topicInfo.lenght < 1) {
+                            return;
+                        }
+
+                        return topicInfo[1];
+                    };
+
+                    span = tracer._startSpan(operationName, {
+                        childOf: parentSpan,
+                        domainName: DomainNames.API,
+                        className: ClassNames.GOOGLEPUBSUB,
+                        disableActiveStart: true,
+                    });
+
+                    const tags = {
+                        [GooglePubSubTags.PROJECT_ID]: getProjectId(),
+                        [GooglePubSubTags.SUBSCRIPTION]: subscription,
+                        [SpanTags.OPERATION_TYPE]: GooglePubSubOperationTypes.PULL,
+                        [SpanTags.SPAN_TYPE]: SpanTypes.GOOGLEPUBSUB,
+                        [SpanTags.TRACE_LINKS]: [span.spanContext.spanId],
+                        [SpanTags.TOPOLOGY_VERTEX]: true,
+                    };
+
+                    span.addTags(tags);
+                    span._initialized();
+                    reachedToCallOriginalFunc = true;
+
+                    const wrappedCallback = (err: any, res: any)  => {
+                        if (!span) {
+                            if (originalCallback) {
+                                originalCallback(err, res);
+                            }
+
+                            return;
+                        }
+
+                        if (err) {
+                            span.setErrorTag(err);
+                        } else if (res && res.receivedMessages) {
+                            const messages = getMessageBody(res.receivedMessages);
+                            if (messages) {
+                                span.addTags({
+                                    [GooglePubSubTags.MESSAGES]: thundraConfig.maskGoogleSubscriptionResponseBody ? undefined
+                                        : JSON.stringify(messages),
```

Comment:
What if we directly assign `messages` without serializing it here as it will be already serialized while reporting?